src/main/java/com/ibm/cloud/sdk/core/security/TokenRequestBasedAuthenticator.java

```diff
@@ -212,11 +225,20 @@ public String getToken() {
     String token;
 
     if (this.tokenData == null || !this.tokenData.isTokenValid()) {
-      // request new token
-      this.tokenData = requestToken();
+      synchronizedRequestToken();
+    } else if (this.tokenData.needsRefresh()) {
+
+      // Kick off background task to refresh token.
+      Thread updateTokenCall = new Thread(new Runnable() {
+        @Override
+        public void run() {
+          synchronizedRequestToken();
```

Comment:
Good call... I think I was worried about multiple threads trying to call that but the refresh time will be updated in `needsRefresh()` (which is synchronized) so that actually shouldn't happen. Changing now.